### PR TITLE
ci: bump PORTREVISION on same-version rebuild

### PR DESCRIPTION
**Pull request by andrebrait**
## Problem

The `sync-ports-fork` job in `release.yml` unconditionally deleted `PORTREVISION` on every release. This meant re-releasing the same tag (or releasing a repackage with no source change) produced the same `PORTVERSION` on the ports fork, giving `pkg` no way to distinguish the new build from the previous one and forcing users to use `pkg install -f` to pick up the rebuild.

pkg version ordering: `4.0.0 < 4.0.0_1 < 4.0.0_2 < 4.0.1` — `PORTREVISION` is the correct "same source, repackaged" counter.

## Change

**Auto-detect rebuild:** when the new `PORTVERSION` equals what is already committed on the ports fork (`pfblockerng/use-github`), increment `PORTREVISION` (`_1`, `_2`, …). When the `PORTVERSION` changes, remove `PORTREVISION` (reset to 0). The decision is per-Makefile, so the devel and stable ports can never cross-trigger.

The logic is extracted into `scripts/portrevision-rebuild.sh` (POSIX sh, three-argument interface: `<old_pv> <old_pr> <new_pv>` → prints new revision integer) so it can be unit-tested independently of the workflow. `release.yml` sparse-checks out `pfBlockerNG/pfBlockerNG` at the release tag (scripts/ only) to call the helper.

## Tests

`tests/shell/portrevision_rebuild_spec.sh` (shellspec, picked up by the existing "shellspec (POSIX sh)" CI job in `test.yml`) pins all branches:

- same version, absent old revision → `1` (first rebuild)
- same version, revision `1` → `2`; revision `2` → `3`
- different version (any old revision) → `0`
- prerelease version rebuild (e.g. `4.0.0.alpha.1`)
- prerelease → stable promotion → `0`
- stable → next prerelease → `0`
- argument count error → exit 2 + usage message

## Accepted side effect

Re-running the release workflow for the **same tag** with no other change will keep incrementing the revision (`_1→_2→_3`). This is intentional per the auto-detect rule — the workflow cannot distinguish "intentional rebuild" from "accidental re-run." This is documented in the commit message and in the docs update below.

## Builder

`build-pkg-portable.py` (`compute_pkgversion()`) already reads `PORTREVISION` from the Makefile and appends `_N` — no builder change is needed or made.

## Docs

- `docs/build-pkg-portable.md`: note explaining that `_N` comes from the Makefile and that the release job keeps it in sync.
- `scripts/README.md`: note in the "Where `.pkg` artifacts land" section covering the auto-detect bump rule and the accepted side effect.

## Files changed

| File | Change |
| --- | --- |
| `.github/workflows/release.yml` | Add sparse checkout of pfBlockerNG scripts; replace unconditional `PORTREVISION=` delete with compare-and-bump via helper |
| `scripts/portrevision-rebuild.sh` | New helper: arithmetic + argument validation (POSIX sh) |
| `tests/shell/portrevision_rebuild_spec.sh` | New shellspec: every branch + edge case |
| `docs/build-pkg-portable.md` | Note on `_N` and the release-time auto-detect bump |
| `scripts/README.md` | Note on the PORTREVISION bump rule and side effect |

---
_Generated by [Claude Code](https://claude.ai/code/session_01D9eEPiZPP6wQUBh762dkTk)_

## Summary by CodeRabbit

* **New Features**
  * Package filenames now automatically reflect `PORTREVISION` via an `_N` suffix for rebuilds when the `PORTVERSION` matches the release tag, and omit/reset the suffix when the version changes.
* **Documentation**
  * Clarified how `PORTREVISION` is incorporated into `.pkg` filenames and how release automation keeps rebuild metadata in sync.
* **Tests**
  * Added shell test coverage for the revision bump/reset logic, including argument validation and edge cases.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -662,14 +671,31 @@ jobs:
...
           MAKEFILE="${PORT_PATH}/Makefile"
 
-          # PORTVERSION = the numeric version; GH_TAGNAME = the actual release tag
-          # (so a real `make package` fetch resolves it); drop PORTREVISION.
+          # Read the current PORTVERSION and PORTREVISION from the Makefile before
+          # editing, so the compare-and-bump helper can decide whether to increment
+          # or reset.  Absent PORTREVISION is treated as 0 by the helper.
+          OLD_PORTVERSION="$(grep -m1 '^PORTVERSION=' "$MAKEFILE" | sed 's/^PORTVERSION=[[:space:]]*//')"
+          OLD_PORTREVISION="$(grep -m1 '^PORTREVISION=' "$MAKEFILE" | sed 's/^PORTREVISION=[[:space:]]*//')" || true
+
```
[the remark this answers is not in the document]
Comment (andrebrait) [reply .github/workflows/release.yml:684]: Applied in 8d3f638. Added an explicit empty-check that fails fast with a clear `::error::` if `PORTVERSION` cannot be parsed from the Makefile.

Note for the record: under GitHub Actions' default `bash -eo pipefail`, the `grep | sed` pipeline already aborts the step when no `PORTVERSION` line matches (grep exits non-zero → pipefail), and an empty value would only bias toward the safe reset-to-`0` path (never a wrong bump). The explicit guard is still worth it for the clearer diagnostic, so applied.

---
🤖 Generated by [Claude Code](https://claude.com/claude-code), posted via @andrebrait's account on their behalf.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by andrebrait** [comment]
@coderabbitai Applied the nitpick in 8d3f638: added `persist-credentials: false` to the sparse `scripts/`-only checkout. That checkout is read-only (it only provides `portrevision-rebuild.sh` to run) and never pushes, so persisting the token is unnecessary exposure.

Both review findings on this PR are now addressed (the `PORTVERSION` fail-fast guard and this credential-persistence hardening).

---
🤖 Generated by [Claude Code](https://claude.com/claude-code), posted via @andrebrait's account on their behalf.

---
_Generated by [Claude Code](https://claude.ai/code)_
